app/services/user_creation_service.rb

```diff
@@ -26,8 +26,13 @@ def call
 
   def update_existing_user
     if full_names_match
+      # Add user to the teams
       @team.__send__(@role.pluralize) << @user
       @team.save!
+
+      # Remove soft delete if their account was previously deleted
+      @user.update(deleted_at: nil) if @user.deleted_at?
```

Comment:
Do we check if `@user` is valid somewhere? I'm guessing not so I think `update!` would be safer. I know that this is probably not the only place where we use `update` where we might want `update!`, but no reason to let this one slip by I think.

Reply:
well spotted I completely forgot the `!` at least the spec was using it ;)